openff/toolkit/typing/engines/smirnoff/parameters.py

```diff
@@ -1812,26 +1812,27 @@ class ParameterHandler(_ParameterAttributeHandler):
 
     """
 
-    _TAGNAME = None  # str of section type handled by this ParameterHandler (XML element name for SMIRNOFF XML representation)
-    _INFOTYPE = None  # container class with type information that will be stored in self._parameters
-    _OPENMMTYPE = None  # OpenMM Force class (or None if no equivalent)
-    _DEPENDENCIES = (
-        None  # list of ParameterHandler classes that must precede this, or None
-    )
-
-    _KWARGS = []  # Kwargs to catch when create_force is called
-    _SMIRNOFF_VERSION_INTRODUCED = (
-        0.0  # the earliest version of SMIRNOFF spec that supports this ParameterHandler
-    )
-    _SMIRNOFF_VERSION_DEPRECATED = (
-        None  # if deprecated, the first SMIRNOFF version number it is no longer used
-    )
+    # str of section type handled by this ParameterHandler (XML element name for SMIRNOFF XML representation)
+    _TAGNAME: Optional[str] = None
+    # container class with type information that will be stored in self._parameters
+    _INFOTYPE: Optional[Any] = None
+    # OpenMM Force class (or None if no equivalent)
+    _OPENMMTYPE: Optional[str] = None
+    # list of ParameterHandler classes that must precede this, or None
+    _DEPENDENCIES: Optional[Any] = None
+
+    # Kwargs to catch when create_force is called
+    _KWARGS: List[str] = []
+    # the earliest version of SMIRNOFF spec that supports this ParameterHandler
+    _SMIRNOFF_VERSION_INTRODUCED = 0.0
+    _SMIRNOFF_VERSION_DEPRECATED = None
+    # if deprecated, the first SMIRNOFF version number it is no longer used
     _MIN_SUPPORTED_SECTION_VERSION = 0.3
     _MAX_SUPPORTED_SECTION_VERSION = 0.3
 
     version = ParameterAttribute()
 
-    @version.converter
+    @version.converter  # type: ignore[no-redef]
```

Comment:
In general I like this PR. However these directives to make mypy happy kinda hurt the readability of the codebase - I'd prefer if there was some way to keep these out/minimize them so that first-time contributors aren't getting confused or getting a failing linter status.